doc: clarify use of deps/icu-small (#43287)
nodejs/node: doc/contributing/maintaining-icu.md

> widely portable and gives applications the same results on all platforms
> and between C/C++ and Java software.

If Node.js is configured to use its built-in ICU,
it uses a strict subset of ICU which is in

> [deps/icu-small](https://github.com/nodejs/node/tree/master/deps/icu-small).

Reviewer: ```suggestion
[deps/icu-small](https://github.com/nodejs/node/tree/HEAD/deps/icu-small).
```
or maybe a relative link would work?
```suggestion
[deps/icu-small](../../deps/icu-small).
```

> A good description of the different ways Node.js can be build with ICU
> support are covered in [api/intl.html](https://nodejs.org/api/intl.html).

Reviewer 2: ```suggestion
A good description of the different ways Node.js can be built with ICU
support is in [api/intl.html](https://nodejs.org/api/intl.html).
```
